fe/fe-core/src/main/java/org/apache/doris/qe/Coordinator.java

```diff
@@ -1252,6 +1253,92 @@ public RowBatch getNext() throws Exception {
         return resultBatch;
     }
 
+
+    // We use a very conservative cancel strategy.
+    // 0. If backends has zero process epoch, do not cancel. Zero process epoch usually arises in cluster upgrading.
+    // 1. If process epoch is same, do not cancel. Means backends does not restart or die.
+    public boolean shouldCancel(List<Backend> currentBackends) {
+        Map<Long, Backend> curBeMap = Maps.newHashMap();
+        for (Backend be : currentBackends) {
+            curBeMap.put(be.getId(), be);
+        }
+        lock();
```

Comment:
use  try (lock()) like RAII in cpp

Reply:
but here we do not have potential thrown exception? can we use try {lock()} finally {unlock();} either?